python/setup.py

```diff
@@ -1,21 +1,84 @@
+import os
+import re
+
 from setuptools import setup, find_packages
 
+
+def find_version(*file_paths):
+    with open(os.path.join(*file_paths), 'rt') as f:
+        version_file = f.read()
+        version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]", version_file, re.M)
+        if version_match:
+            return version_match.group(1)
+    raise RuntimeError("Unable to find version string.")
+
+
+version = find_version('fusion_engine_client', '__init__.py')
+
+tools_requirements = set([
+    'argparse-formatter>=1.4',
+    'gpstime>=0.6.2',
+])
+
+display_requirements = set([
+    'colorama>=0.4.4',
+    'palettable>=3.3.0',
+    'plotly>=4.0.0',
+    'pymap3d>=2.4.3',
+]) | tools_requirements
+
+all_requirements = tools_requirements | display_requirements
+
 setup(
     name='fusion-engine-client',
-    version='v1.15.0',
+    version=version,
+    description='Point One FusionEngine Library',
+    long_description="""\
+Point One FusionEngine protocol support for real-time interaction and control, plus post-processing data analysis tools.
+
+See https://github.com/PointOneNav/fusion-engine-client for full details. See https://pointonenav.com/docs/
+for the latest FusionEngine message specification.
+""",
+    long_description_content_type='text/markdown',
+    author='Point One Navigation',
+    author_email='support@pointonenavigation.com',
+    license='MIT',
+    classifiers=[
+        'Development Status :: 5 - Production/Stable',
+        'Intended Audience :: Developers',
+        'Intended Audience :: End Users/Desktop',
+        'Intended Audience :: Science/Research',
+        'License :: OSI Approved :: MIT License',
+        'Operating System :: MacOS :: MacOS X',
+        'Operating System :: Microsoft :: Windows',
+        'Operating System :: POSIX',
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.8',
```

Comment:
should this include 3.7?

Reply:
Technically we have it set to require >3.7. We have not tested on anything earlier than 3.8. We can if we want to in the future, but 3.8 was released in 2019 and 3.7 support ended in 2020 so this is probably safe.

Reply:
I updated the README to say 3.8 or later instead of 3.7